rktl_sim/src/simulator/car.py

```diff
@@ -113,14 +120,42 @@ def get_velocity(self):
         heading = p.getEulerFromQuaternion(orient)[2]
         r_inv = np.array([[math.cos(heading), -math.sin(heading), 0.],
                           [math.sin(heading), math.cos(heading), 0.],
-                         [0., 0., 1.]], dtype=np.float)
+                          [0., 0., 1.]], dtype=np.float)
         linear = r_inv @ linear
-        return (linear, angular)
+        return linear, angular
 
     def reset(self, pos, orient):
+        """
+        resets the system state and the model configuration for the car
+        uses the position and the orientation to reset the cars position
+        @param pos: the new position of the car
+        @param orient: the new orientation of the car
+        """
+        # save the car pos and orient:
+        self.pos = pos
+        self.orient = orient
+
+        # system state
         self._v_rear = 0.0
         self._psi = 0.0
 
+        # model configuration
+        p.resetBasePositionAndOrientation(
+            self.id, [0., 0., pos[2]], p.getQuaternionFromEuler([0., 0., 0.]))
+
+        self.joint_ids = (1, 0, 2)  # X, Y, W
         p.resetJointState(self.id, self.joint_ids[0], targetValue=pos[0])
         p.resetJointState(self.id, self.joint_ids[1], targetValue=pos[1])
         p.resetJointState(self.id, self.joint_ids[2], targetValue=orient[2])
+
+        self.cmd = None
+
+    def check_overlap(self, pos):
+        """
+        returns whether the two positions of the car will overlap
+        @param pos: the position of the other car
+        @return: bolean if they overlap (true = overlap)
+        """
+        dist = math.sqrt(
+            (pos[0] - self.pos[0]) * (pos[0] - self.pos[0]) + (pos[0] - self.pos[1]) * (pos[1] - self.pos[1]))
+        return dist < self._LENGTH
```

Comment:
Out of curiosity what objects would we have that would be larger than a car? 
If so, it would make sense to create a general object class with general methods such as reset, step, get velocity, and so on and have the car, ball and other objects extend it for simplicity. That way I can have a general check_overlap function in that clas that would check the distance from each object using the largest length of the car.

(I changed the property from self.pos to self.init_pos